tests/rptest/tests/audit_log_test.py

```diff
@@ -667,8 +667,8 @@ def test_drain_on_audit_disabled(self):
         _ = self.find_matching_record(
             partial(AuditLogTestsAppLifecycle.is_lifecycle_match,
                     "Audit System", False),
-            lambda record_count: record_count == 3,
-            "One stop event observed for shutdown node")
+            lambda record_count: record_count >= 3,
+            "One or more stop event(s) observed per node")
```

Comment:
This seems non ideal because we don't ensure each node emitted at least one stop event (ie. this test could pass if one node produced two and another produced 0).

Ideally we assert each node emitted at least one shutdown record. It does seem like a bug to have multiple shutdown audit events when the node only shuts down once, but 🤷

Reply:
Unfortunately theres no way to tie a node to an event since the node id is not a part of the message payload, example below:
```
{'category_uid': 6, 'class_uid': 6002, 'metadata': {'product': {'name': 'Redpanda', 'vendor_name': 'Redpanda Data, Inc.', 'version': 'v23.3.1-rc1-228-gc378b3d489'}, 'version': '1.0.0'}, 'severity_id': 1, 'time': 1701276934847, 'type_uid': 600204, 'activity_id': 4, 'app': {'feature': {'name': 'Audit System'}, 'name': 'Redpanda', 'vendor_name': 'Redpanda Data, Inc.', 'version': 'v23.3.1-rc1-228-gc378b3d489'}}
```
Furthermore there is no difference between "shutdown" and "audit disabled" they send the same events onto the audit log. So what we are seeing in the failed test were 5 total identical events, 3 coming from disabling auditing and 2 coming from shutting down redpanda gracefully.

Agreed, ideally it would be nice to know which node emitted which event, then we could assert that no duplicates exist and all of that jazz.  I guess a solution to this would be to differentiate between stop and disable audit and also add node id or ip to the payload. @michael-redpanda could maybe expand here.

Reply:
To my point #1 above, that `is_lifecycle_match`  method is being called to look for the audit logging related app lifecycle message.

Reply:
Now that it's rebased with the mapper fix, you should only see 3 events, can that be tested?

Reply:
done, resolving